### ISSUE TITLE
eRegistry server doesn't work

### ISSUE THREAD

**Issue by ifoche** [EyeSeeTea/dhis2-android-trackercapture#141]

**Comment by ifoche** [comment]
The alpha tester reports 2 different problems with eRegistry server (this is a fork that's not exactly a normal DHIS2 server, but we have to make the app work also with this fork if possible). 

1) Logged in as a user assigned to a single org unit to https://eregistry.dhis2.org/demo. A message comes up that I am not assigned to a unit, or do not have sufficient permission. 
The same message occurs if I log in as the admin (user: admin password: District9).

2) (I guess that to reach 2, 1 must be already solved so don't spend too much time on it) Crashes consistently while trying to enroll into the MCH program (eregistry.dhis2.org/demo). Probably the app doesn't like this forked version of Tracker, but it's worth finding out why not.

**Comment by idelcano** [comment]
@ifoche 

1. The admin user pull the follow elements:
Organisation unit:
"GD7TowwI46c"	"Palestine"
"tdTZDOYNwgD"	"عيادة إماتين"

Programs:
WSGAb5XwJ3Y

This program is not assigned to the pulled organisation units, and it's removed after finish the sync.
The user haven't programs after the pull, and shows the reported message.

I tested the admin user  and works ok  after adding the "Palestine" orgunit to the WSGAb5XwJ3Y program.

I tested the user  username:  User password: Testing123 and works ok.

2. The crashes are provocated by a null program rule variable elements, and this is solved in the PR.
